crates/cheatcodes/src/inspector.rs

```diff
@@ -1556,6 +1609,33 @@ impl Cheatcodes {
         }
     }
 
+    #[cold]
+    fn meter_gas_record<DB: DatabaseExt>(
+        &mut self,
+        interpreter: &mut Interpreter,
+        ecx: &mut EvmContext<DB>,
+    ) {
+        println!(
+            "{:?} - {:?} @ {}",
+            interpreter.instruction_result,
+            interpreter.gas.spent(),
+            ecx.journaled_state.depth()
+        );
+
+        if self.gas_metering.last_recorded_depth != ecx.journaled_state.depth() {
+            self.gas_metering.last_recorded_frame = 0;
+            self.gas_metering.last_recorded_depth = ecx.journaled_state.depth();
+        }
+
+        if matches!(interpreter.instruction_result, InstructionResult::Continue) {
+            self.gas_metering
```

Comment:
do we need to push frame here? I think we can just directly sum up the gas usage?

Reply:
likely not, will likely end up removing it - mostly using it now for debugging purposes